shell/common/node_bindings.cc

```diff
@@ -689,6 +745,35 @@ void NodeBindings::StartPolling() {
   UvRunOnce();
 }
 
+void NodeBindings::SetAppCodeLoaded() {
+  app_code_loaded_ = true;
+}
+
+void NodeBindings::JoinAppCode() {
+  // We can only "join" app code to the main thread in the browser process
+  if (browser_env_ != BrowserEnvironment::kBrowser) {
+    return;
+  }
+
+  auto* browser = Browser::Get();
+  node::Environment* env = uv_env();
+
+  if (!env)
+    return;
+
+  v8::HandleScope handle_scope(env->isolate());
+  // Enter node context while dealing with uv events.
+  v8::Context::Scope context_scope(env->context());
+
+  // Pump the event loop until we get the signal that the app code has finished
+  // loading
+  while (!app_code_loaded_ && !browser->is_shutting_down()) {
+    int r = uv_run(uv_loop_, UV_RUN_NOWAIT);
```

Comment:
does `UV_RUN_NOWAIT` mean we busyloop here? perhaps instead we should consider using a `uv_cond_t` instead of busywaiting on a bool?

Reply:
`UV_RUN_NOWAIT` means it doesn't hang `uv_run` if there is no work to do, we just end up calling `uv_run` over and over again instead.  What would you want to see instead of this style of loop pump?

Reply:
We do want to hang when there's no work to do though, right? That's e.g. when we're waiting on fs/network/etc., we definitely don't want to spin while that's happening, as we'll eat 100% cpu just checking that app_code_loaded_ is false.

Does this Just Work if you set it to UV_RUN_DEFAULT? I'd imagine that app_code_loaded_ won't change if the UV loop isn't doing any work, so we're likely safe to wait.

Reply:
`UV_RUN_DEFAULT` will run the loop until node terminates (by definition).  I could swap it for `UV_RUN_ONCE` so that we aren't spinlooping at 100% CPU but also I don't think the 100% CPU spinloop is super possible because `appCodeLoaded` is called pretty quickly.  Although technically speaking someone _could_ `await` a network thing in the main process they'd be pretty stupid to do so.

In fact we should probably document that 🤔

Reply:
I think it's not unreasonable to await for IO things at top level if you really do want to delay app load. Just out of interest, how many times does this loop run if you do a simple `await fs.readFile`? what about for `fs.readFile` of something that's not in disk cache (`sudo purge` to drop disk caches)?

UV_RUN_ONCE seems like the right choice here.